### PR TITLE
Surface package changelogs on docs site

### PR DESCRIPTION
**Pull request by masenf**
### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### Description

This PR adds changelog discovery and rendering to the docs site, making package changelogs accessible at `/docs/changelog/` instead of requiring users to visit GitHub releases.

**What changed:**

1. **New `reflex_docs/changelogs.py` module** — Discovers and normalizes changelogs:
   - `discover_repo_changelogs()` finds towncrier-managed changelogs in the monorepo (repo root + `packages/*/CHANGELOG.md`)
   - `find_distribution_changelog()` locates changelogs shipped with installed distributions (e.g., `reflex-enterprise`)
   - `discover_changelogs()` combines both sources, ordering `reflex` first then alphabetically
   - `normalize_changelog()` ensures consistent H1 headings across different changelog formats (towncrier vs. Keep-a-Changelog)

2. **Changelog rendering in docs** — Integrated into the docs pipeline:
   - New `handle_changelog_doc()` handler normalizes changelog markdown and limits TOC to version headings (H2 and above)
   - Changelogs are registered as virtual docs under `docs/changelog/` with the main `reflex` changelog at the section index
   - Updated footer link to point to `/changelog/` instead of GitHub releases

3. **Sidebar navigation** — Added changelog section to the docs sidebar:
   - New `get_sidebar_items_changelog()` dynamically generates sidebar items from discovered packages
   - Changelog routes are treated as API reference content for sidebar styling

4. **Helper function** — Added `render_markdown_with_toc()` to extract heading structure while rendering markdown

5. **Documentation** — Updated `CONTRIBUTING.md` to explain where changelogs are published

### Tests

- Added comprehensive test suite in `docs/app/tests/test_changelogs.py` covering:
  - Repo changelog discovery (monorepo structure)
  - Distribution changelog discovery (installed packages)
  - Changelog normalization (towncrier and Keep-a-Changelog formats)
  - Page title generation
  - Edge cases (missing distributions, stale file records, no file metadata)
- Added route validation test in `test_routes.py` to ensure all discovered changelogs are served
- Added section naming test in `test_agent_files.py` for root-level markdown docs

All tests pass with the changes.

### Related issues

Closes the need for users to navigate to GitHub releases for changelog information.

https://claude.ai/code/session_01ToXo8Yg1VTuuT2jgBqK6T7

### CODE REVIEW

File: docs/app/reflex_docs/changelogs.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+def changelog_page_title(package: str) -> str:
+    """Return the display title for a package changelog page.
+
+    Args:
+        package: The package name.
+
+    Returns:
+        The page title.
+    """
+    return "Reflex Changelog" if package == "reflex" else f"{package} Changelog"
```
[the remark this answers is not in the document]
Comment (masenf) [reply docs/app/reflex_docs/changelogs.py:89]: Intentional: these pages document distribution packages, so the title keeps the literal PyPI name (`reflex-base`, `reflex-components-markdown`) rather than a title-cased rendering that would obscure the actual package identity. Only the main framework page gets the branded "Reflex Changelog" title.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: docs/app/reflex_docs/changelogs.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        title: The canonical page title.
+
+    Returns:
+        The normalized markdown.
+    """
+    lines = source.lstrip().splitlines()
+    if lines and lines[0].startswith("# "):
+        del lines[0]
+    body = "\n".join(lines).strip("\n")
+    return f"# {title}\n\n{body}\n"
```
[the remark this answers is not in the document]
Comment (masenf) [reply docs/app/reflex_docs/changelogs.py:111]: Acknowledged but leaving as-is: every changelog rendered here is machine-generated by towncrier (configured in this repo) or Keep-a-Changelog tooling, both of which emit ATX headings. If a setext file ever shows up, the symptom is an obvious duplicate heading on the page, easy to fix then — handling it preemptively is dead code today.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: docs/app/reflex_docs/changelogs.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    """
+    try:
+        dist = distribution(package)
+    except PackageNotFoundError:
+        return None
+    for file in dist.files or ():
+        if file.name == "CHANGELOG.md":
+            path = Path(str(dist.locate_file(file)))
+            if path.is_file():
+                return path
```
[the remark this answers is not in the document]
Comment (masenf) [reply docs/app/reflex_docs/changelogs.py]: Good catch — wheels can vendor third-party changelogs in bundled assets. Fixed in ff4d8d32: candidates are now sorted by path depth so the shallowest (package-level) record wins, with a regression test.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by adhami3310 — APPROVED** [review]
